### PR TITLE
send_null pattern 

### PR DESCRIPTION
**Pull request by silverCase**
issue #148

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#197](https://github.com/yegor256/aibolit/issues/197) is assigned to @lyriccoder[/z](https://www.0crat.com/u/lyriccoder), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/bb2653ca-7539-4529-9bf2-a73b39b6c1e8); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Job [#197](https://github.com/yegor256/aibolit/issues/197) is already in scope

**Comment by 0crat** [comment]
GitHub issue is still open, won't close

File: test/patterns/send_null/test_send_null.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        lines = self.method_send_null_finder.value(Path(self.dir_path, 'Configuration.java'))
+        self.assertEqual(lines, [442, 379, 638, 656, 830, 866, 1362, 2393, 2988, 3080, 3492])
+
+    def test_no_null_methods(self):
+        lines = self.method_send_null_finder.value(Path(self.dir_path, 'FillContent.java'))
+        self.assertEqual(lines, [])
+
+    def test_simple_invocation(self):
+        lines = self.method_send_null_finder.value(Path(self.dir_path, 'FJIterateTest.java'))
+        self.assertEqual(lines, [493])
```
Comment (lyriccoder) [inline test/patterns/send_null/test_send_null.py:25]: Check the following cases if you didn't test it:

`a.method_call(2, b.method_call(null))`
`a.method_call().method_call(b).method_call(null)`
Check, if we can pass a ternary operator:
`a.method_call(5>8? 5:null)`
`a.method_call(5>8? null:5)`
Check Constructor:
`new Object(null)`

Comment (silverCase) [reply test/patterns/send_null/test_send_null.py:25]: Will do, thanks

**Comment by acheshkov** [comment]
@silverCase resolve conflict, please

File: test/patterns/send_null/test_send_null.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    def test_more_method_invocations(self):
+        lines = self.method_send_null_finder.value(Path(self.dir_path, 'SequenceFile.java'))
+        self.assertEqual(lines, [1186, 1201, 1217, 3285, 3298, 3367, 3537, 3550])
+
+    def test_constructor_send_null(self):
+        lines = self.method_send_null_finder.value(Path(self.dir_path, 'Constructor.java'))
+        self.assertEqual(lines, [5, 14, 15, 18])
```
Comment (lyriccoder) [inline test/patterns/send_null/test_send_null.py:33]: You forgot  to add 16, 17 lines:

```
       doSomething(myString = ( ( myString != 5) ? null : myString ), obj);
        doSomething(myString = ( ( myString != 5) ? myString.toLowerCase() : null), obj);
```

Comment (silverCase) [reply test/patterns/send_null/test_send_null.py]: Done.

**Comment by 0crat** [comment]
@lyriccoder[/z](https://www.0crat.com/u/lyriccoder) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by lyriccoder** [comment]
@acheshkov you can merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@silverCase take a look! type checking errors

**Comment by acheshkov** [comment]
@rultor merge

**Pull request merged**